src/util/options.schema.json

```diff
@@ -738,6 +738,12 @@
               "type": "boolean",
               "default": true
             },
+            "strengthening": {
+              "title": "ana.apron.strengthening",
+              "description": "Apply strengthening in join for extra precision with heterogeneous environments. Expensive!",
+              "type": "boolean",
+              "default": false
+            },
```

Comment:
Is it always sound to apply this? I am not 100% convinced if we do not rely on not having strengthening for some heterogeneous environments now?

Reply:
The algorithm of strengthening contains a soundness check within. It just tries reasserting all constraints from both arguments and checks if the result is still sound.
None of the privatization stuff if affected by this anyway because there the environments are homogeneous.

Reply:
Ok! Since we now have that option, should we maybe add a test for it then?

Reply:
Good idea! I don't remember what the original examples were, where this was needed (possibly some initial write-centered Apron thing), but I found an example where `protection` needs it to keep precision.